dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaAPIHostsTaskTest.java

```diff
@@ -168,6 +126,108 @@ public void clustersWereNotInitialized_throwsException() throws Exception {
     verifyNoWrites();
   }
 
+  @Test
+  public void doRun_clouderaReturns4xx_throwsCriticalException() throws Exception {
+    initClusters(ClouderaClusterDTO.create("id1", "first-cluster"));
+    mockClouderaClusterAPIResponse("first-cluster", "[]", HttpStatus.SC_BAD_REQUEST);
+
+    MetadataDumperUsageException exception =
+        assertThrows(MetadataDumperUsageException.class, () -> task.doRun(context, sink, handle));
+
+    assertTrue(exception.getMessage().contains("Cloudera Error:"));
+  }
+
+  @Test
+  public void doRun_clouderaReturns5xx_throwsCriticalException() throws Exception {
+    initClusters(ClouderaClusterDTO.create("id1", "first-cluster"));
+    mockClouderaClusterAPIResponse("first-cluster", "[]", HttpStatus.SC_INTERNAL_SERVER_ERROR);
+
+    MetadataDumperUsageException exception =
+        assertThrows(MetadataDumperUsageException.class, () -> task.doRun(context, sink, handle));
+
+    assertTrue(exception.getMessage().contains("Cloudera Error:"));
+  }
+
+  @Test
+  public void doRun_clouderaReturnsInvalidJsonFormat_throwsCriticalException() throws Exception {
+    initClusters(ClouderaClusterDTO.create("id1", "first-cluster"));
+    mockClouderaClusterAPIResponse("first-cluster", "[}");
+
+    MetadataDumperUsageException exception =
+        assertThrows(MetadataDumperUsageException.class, () -> task.doRun(context, sink, handle));
+
+    assertTrue(exception.getMessage().contains("Cloudera Error:"));
+  }
+
+  private void initClusters(ClouderaClusterDTO... clusters) {
+    handle.initClusters(Arrays.asList(clusters));
+  }
+
+  private CloseableHttpResponse mockClouderaClusterAPIResponse(
+      String clusterName, String itemsJsonLine) throws IOException {
+    return mockClouderaClusterAPIResponse(clusterName, itemsJsonLine, HttpStatus.SC_OK);
+  }
+
+  private CloseableHttpResponse mockClouderaClusterAPIResponse(
```

Comment:
It was agreed do not use `mock` for Http testing, please use WireMock server.

Example https://github.com/google/dwh-migration-tools/pull/678